cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -181,25 +181,18 @@ func GetNodeRegistration(kubeconfigFile string, client clientset.Interface, node
 		ok                     bool
 		missingAnnotationError = errors.Errorf("node %s doesn't have %s annotation", nodeName, constants.AnnotationKubeadmCRISocket)
 	)
-	if features.Enabled(clusterCfg.FeatureGates, features.NodeLocalCRISocket) {
-		_, err = os.Stat(filepath.Join(constants.KubeletRunDirectory, constants.KubeletInstanceConfigurationFileName))
-		if os.IsNotExist(err) {
-			criSocket, ok = node.ObjectMeta.Annotations[constants.AnnotationKubeadmCRISocket]
-			if !ok {
-				return missingAnnotationError
-			}
-		} else {
-			kubeletConfig, err := readKubeletConfig(constants.KubeletRunDirectory, constants.KubeletInstanceConfigurationFileName)
-			if err != nil {
-				return errors.Wrapf(err, "node %q does not have a kubelet instance configuration", nodeName)
-			}
-			criSocket = kubeletConfig.ContainerRuntimeEndpoint
-		}
-	} else {
-		criSocket, ok = node.ObjectMeta.Annotations[constants.AnnotationKubeadmCRISocket]
-		if !ok {
+	criSocket, ok = node.Annotations[constants.AnnotationKubeadmCRISocket]
+	if !ok {
+		configFilePath := filepath.Join(constants.KubeletRunDirectory, constants.KubeletInstanceConfigurationFileName)
+		_, err := os.Stat(configFilePath)
+		if os.IsNotExist(err) && !features.Enabled(clusterCfg.FeatureGates, features.NodeLocalCRISocket) {
 			return missingAnnotationError
```

Comment:
previously the error was strictly returned for a missing annotation but in this case both the annotation and instance config are missing. maybe better to update the error to reflect that.

e.g.
```
return errors.Errorf("node %s doesn't have %s annotation, or instance config %s", nodeName, constants.AnnotationKubeadmCRISocket, configFilePath)
```